### DISCUSSION TITLE
Is response_filter behaving as expected/intended?

### DISCUSSION THREAD

**Discussion by filipebezerra** [sh4nks/flask-caching#231] [Ideas]
> I'm using Flask-Caching Version: 1.9.0

I'm trying to figure out how `response_filter` behaves, it looks like it doens't prevent 500 responses. For instance let's suppose we setup redis backend and the application weren't able to connection because something bad happened to connect to redis instance, so it throws an `redis.exceptions.ConnectionError`, in the way the decorator `memoize` was written the `response_filter` calllable will end up not having the chance to handle this connection error and the application will throw the 500 error to the client.

I'm right here folks?

**Comment by sh4nks** [comment]
No, in this case it can't prevent 500 responses - you'd have to check in your app if your caching backend works or not.

**Comment by filipebezerra** [comment]
I see, in this case is it possible to extend `response_filter` to support that case? I could work with that if it is possible.

I agree with you that is on the application developer the responsability to check if the Redis server is available or not, but I think it's flask-caching responsability to not throw such kind of exception, handling it and maybe throwing a specific error or even a warning log.

It will be more handy to use `response_filter` to avoid that application crashes when was trying to cache the response.

**Comment by sh4nks** [comment]
If you have got an idea on how to solve that, feel free to open a new PR :)
I'll try to review it as soon as possible.
